test/clj/puppetlabs/trapperkeeper/services/webserver/jetty9_service_test.clj

```diff
@@ -729,7 +712,34 @@
           (let [response (http-client-common/get async-client "http://localhost:8080/hello" {:as :text})]
             @in-request-handler
             (tk-app/stop app)
-            (is (not (nil? (:error @response))))))))))
+            (is (not (nil? (:error @response)))))))))
+
+  (testing "tk app can still restart even if stop timeout expires"
+    (let [in-request-handler (promise)
+          sleepy-service (tk-core/service
+                          [[:WebserverService add-ring-handler]]
+                          (init [this context]
+                                (add-ring-handler
+                                 (fn [_]
+                                   (deliver in-request-handler true)
+                                   (Thread/sleep 2000)
```

Comment:
Rather than just putting a sleep here, might it be better to just have a line here that waits on a promise that gets delivered from the main thread just after the `tk-app/restart` call occurs?  That way, we wouldn't have the possible race condition where this sleep might finish and the response start to be delivered before Jetty fully gets into its `stop` sequence.